lms/static/js/utils/dropdown.js

```diff
@@ -0,0 +1,195 @@
+var edx = edx || {},
+    Dropdown = (function() {
+
+        var dropdown = {
+
+            init: function(parent) {
+                dropdown.optsSet();
+
+                if (dropdown.opts.button.length && dropdown.opts.menu.length) {
+
+                    if (parent) {
+                        dropdown.opts.page = $(parent);
+                    }
+
+                    dropdown.listenForClick();
+                    dropdown.listenForKeypress();
+                }
+            },
+
+            optsSet: function() {
+                dropdown.opts = {
+                    page: $(document),
+                    midpoint: ($(window).width() / 2),
+                    button: $('.button-more.has-dropdown'),
+                    button_active: 'is-active',
+                    menu: $('.dropdown-menu'),
+                    menu_inactive: 'is-hidden',
+                    menu_active: 'is-visible',
+                    menu_align: 'align-',
+                    text_align: 'control-text-align'
+                };
+            },
+
+            listenForClick: function() {
+                dropdown.opts.button.on('click', function() {
+                    dropdown.closeDropdownMenus(); // close any open menus
+                    dropdown.openDropdownMenu($(this)); // then open the chosen menu
+                });
+
+                dropdown.opts.page.on('click', function() {
+                    dropdown.closeDropdownMenus();
+                });
+            },
+
+            handlerIsAction: function(key, menu, focused, e) {
+                e.preventDefault();
```

Comment:
You should only preventDefault if you actually handle the keystroke. Same for all of these methods.

You can see this in the deleted my_courses_dropdown.js.

You should also preventDefault in the click handlers.